.github/workflows/link-check.yml

```diff
@@ -0,0 +1,53 @@
+name: Build and check Links
+
+on:
+  push:
+    branches:
+      - main
+
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  build_and_serve:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+
+      - name: Install Dependencies
+        run: yarn install --frozen-lockfile
+
+      - name: Build Next.js App
+        run: yarn build
+
+      - name: Serve App Locally
+        run: yarn start &
+
+      - name: Wait for App to Start
+        run: sleep 20
+
+  linkChecker:
+    runs-on: ubuntu-latest
+    needs: build_and_serve
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
```

Comment:
Add steps to run our app locally.

Reply:
see comment [here](https://github.com/json-schema-org/website/pull/297#discussion_r1511154012)